core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -406,6 +406,15 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
   def getBoolean(key: String, defaultValue: Boolean): Boolean = {
     getOption(key).map(_.toBoolean).getOrElse(defaultValue)
   }
+  /** Get a parameter as a Option[Boolean] */
+  def getOptionBoolean(key: String): Option[Boolean] = {
```

Comment:
@xanec I drifted away from this because I wasn't sure about the behavior in case of non matching types(exception), but then noticed that `getDouble` and `getLong` also throw exception and do not default to the default value (bug or design ?).
So if they don't handle exceptions, then I can use what you suggested